app/services/SchemeSearchService.scala

```diff
@@ -58,4 +58,21 @@ class SchemeSearchService @Inject()(listSchemesConnector: ListOfSchemesConnector
         case _ => List.empty[SchemeDetails]
       }
   }: Future[List[SchemeDetails]]
+
+  private val filterPspSchemesByPstr: (String, List[SchemeDetails]) => List[SchemeDetails] =
+    (searchText, list) => list.filter(_.pstr.contains(searchText))
```

Comment:
What is the benefit of declaring a val that is a function over a def? So far as I can tell, this just takes a bit of extra thinking over say: 
`def filterPspSchemesByPstr(searchText: String, list: List[SchemeDetails]): List[SchemeDetails] = 
  list.filter(_.pstr.contains(searchText))`

Reply:
Is there any functional difference?

Reply:
In reality the filter can be moved into the search def - that will make it easier to follow I think. You're right it doesn't really need to be a val

Reply:
It wasn't a criticism as such, more a question for my own understanding. Happy either way really.

Reply:
No I think you are right. I am refactoring now, will push changes shortly

Reply:
Refactored as per comment